### PR TITLE
Test for shopping cart functionality

### PR DESCRIPTION
**Pull request by RedArtelerist**
image

### CODE REVIEW

File: src/test/java/maksym/fedorenko/bookstore/controller/ShoppingCartControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@Sql(scripts = "classpath:database/auth/add-default-users.sql")
+@Sql(scripts = {
+        "classpath:database/shopping_cart/delete-all-items.sql",
+        "classpath:database/auth/delete-users.sql"},
+        executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD
+)
+@SqlMergeMode(SqlMergeMode.MergeMode.MERGE)
+public class ShoppingCartControllerTest {
+    protected static MockMvc mockMvc;
```
Comment (NazarSmith228) [inline src/test/java/maksym/fedorenko/bookstore/controller/ShoppingCartControllerTest.java]: why protected?

File: src/test/java/maksym/fedorenko/bookstore/controller/ShoppingCartControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+                .hasFieldOrPropertyWithValue("quantity", 1);
+    }
+
+    @Test
+    @DisplayName("Get shopping cart by non-existent user")
+    @WithMockUser(username = "another_user@gmail.com", roles = "USER")
+    @Sql(scripts = "classpath:database/shopping_cart/add-one-item.sql")
+    public void getUserCart_WithNonExistentUser_BadRequest() throws Exception {
+        MvcResult result = mockMvc.perform(get("/api/cart")
+                        .contentType(MediaType.APPLICATION_JSON))
```
Comment (NazarSmith228) [inline src/test/java/maksym/fedorenko/bookstore/controller/ShoppingCartControllerTest.java]: this is a GET - why do you need content type?

File: src/test/java/maksym/fedorenko/bookstore/controller/ShoppingCartControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+        objectMapper = new ObjectMapper().registerModule(new JavaTimeModule());
+    }
+
+    @Test
+    @DisplayName("Get shopping cart by existing user")
+    @WithMockUser(username = "user@gmail.com", roles = "USER")
+    @Sql(scripts = "classpath:database/shopping_cart/add-one-item.sql")
+    public void getUserCart_WithExistingUser_Success() throws Exception {
+        MvcResult result = mockMvc.perform(get("/api/cart")
+                        .contentType(MediaType.APPLICATION_JSON))
```
Comment (NazarSmith228) [inline src/test/java/maksym/fedorenko/bookstore/controller/ShoppingCartControllerTest.java]: this is a GET - why do you need content type?

File: src/test/java/maksym/fedorenko/bookstore/controller/ShoppingCartControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+                .andExpect(status().isNotFound())
+                .andReturn();
+
+        ErrorResponseWrapper error = objectMapper.readValue(
+                result.getResponse().getContentAsString(), ErrorResponseWrapper.class
+        );
+
+        assertThat(error.details()).startsWith("Can't find cart for user with email");
```
Comment (NazarSmith228) [inline src/test/java/maksym/fedorenko/bookstore/controller/ShoppingCartControllerTest.java:95]: why not asserting all fields?

File: src/test/java/maksym/fedorenko/bookstore/controller/ShoppingCartControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+                        .content(objectMapper.writeValueAsString(requestDto)))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        CartDto actual = objectMapper.readValue(
+                result.getResponse().getContentAsString(),
+                CartDto.class
+        );
+
+        assertThat(actual).isNotNull()
```
Comment (NazarSmith228) [inline src/test/java/maksym/fedorenko/bookstore/controller/ShoppingCartControllerTest.java]: minor 
```suggestion
        assertThat(actual)
```

File: src/test/java/maksym/fedorenko/bookstore/repository/BookRepositoryTest.java [hunk 6]
```diff
@@ -5,67 +5,30 @@
...
 import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
 import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
-import org.springframework.core.io.ClassPathResource;
 import org.springframework.data.domain.Page;
 import org.springframework.data.domain.PageRequest;
-import org.springframework.jdbc.datasource.init.ScriptUtils;
+import org.springframework.test.context.jdbc.Sql;
 
 @DataJpaTest
 @AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
```
Comment (NazarSmith228) [inline src/test/java/maksym/fedorenko/bookstore/repository/BookRepositoryTest.java:21]: do you really need this? `NONE` means use autoconfigured datasource, which is H2 here

therefore I guess you can skip this annotation - pls verify

**Applicable to all such tests**

File: src/test/java/maksym/fedorenko/bookstore/service/ShoppingCartServiceTest.java [hunk 7]
```diff
@@ -0,0 +1,267 @@
...
+        assertThatThrownBy(() -> shoppingCartService.getUserCart(authentication))
+                .isInstanceOf(EntityNotFoundException.class)
+                .hasMessage("Can't find cart for user with email: %s".formatted(username));
+    }
+
+    @Test
+    @DisplayName("Add valid item to cart")
+    void addCartItem_ValidRequestDto_ShouldReturnCartDto() {
+        final String username = "user@gmail.com";
+        final CreateCartItemRequestDto requestDto = new CreateCartItemRequestDto(1L, 2);
```
Comment (NazarSmith228) [inline src/test/java/maksym/fedorenko/bookstore/service/ShoppingCartServiceTest.java:89]:
```suggestion
        String username = "user@gmail.com";
        CreateCartItemRequestDto requestDto = new CreateCartItemRequestDto(1L, 2);
```

**Applicable to all places**

**Review by NazarSmith228 — APPROVED** [review]
LGTM, but pls pay attention to my remarks

File: src/test/java/maksym/fedorenko/bookstore/service/ShoppingCartServiceTest.java (hunk 7 above)
Comment (RedArtelerist) [reply src/test/java/maksym/fedorenko/bookstore/service/ShoppingCartServiceTest.java:89]: checkstyle swear that the ad is far from the first use

**Pull request merged**